pkgs/applications/networking/instant-messengers/psi-plus/default.nix

```diff
@@ -3,37 +3,117 @@
+      sha256 = "0cj811qv0n8xck2qrnps2ybzrpvyjqz7nxkyccpaivq6zxj6mc12";
+    };
+
+    cmakeFlags = [
+      "-DENABLE_PLUGINS=ON"
+    ];
 
-  cmakeFlags = [
-    "-DENABLE_PLUGINS=ON"
-  ];
-
-  nativeBuildInputs = [ cmake qttools ];
-
-  buildInputs = [
-    qtbase qtmultimedia qtx11extras qtwebengine
-    libidn qca-qt5 libXScrnSaver hunspell
-    libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
-    usrsctp
-  ];
-
-  meta = with lib; {
-    homepage = "https://psi-plus.com";
-    description = "XMPP (Jabber) client";
-    maintainers = with maintainers; [ orivej misuzu ];
-    license = licenses.gpl2Only;
-    platforms = platforms.linux;
+    nativeBuildInputs = [ cmake qttools ];
+
+    buildInputs = [
+      qtbase qtmultimedia qtx11extras qtwebengine
+      libidn qca-qt5 libXScrnSaver hunspell
+      libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
+      usrsctp
+    ];
+
+    meta = with lib; {
+      homepage = "https://psi-plus.com";
+      description = "XMPP (Jabber) client";
+      maintainers = with maintainers; [ orivej misuzu ];
+      license = licenses.gpl2Only;
+      platforms = platforms.linux;
+    };
   };
-}
+
+  psi-plus-wrapped =
+    let
+      gstLibSubDir = x: "${x}/lib/gstreamer-1.0";
+      gstEnvVarName = "GST_PLUGIN_SYSTEM_PATH_1_0";
+
+      gstDependencies = [
+        gstreamer
+        gst-plugins-base
+        gst-plugins-good
+      ];
+
+      checkDirExistence = dir: ''
+        if [[ ! -d ${escapeShellArg dir} ]]
+        then
+          >&2 printf 'Directory "%s" not found!\n' ${escapeShellArg dir}
+          exit 1
+        fi
+      '';
+
+      checkExecutableExistence = exe: ''
+        if [[ ! -f ${escapeShellArg exe} ]] \
+        || [[ ! -r ${escapeShellArg exe} ]] \
+        || [[ ! -x ${escapeShellArg exe} ]]
+        then
+          >&2 printf 'Executable "%s" not found!\n' ${escapeShellArg exe}
+          exit 1
+        fi
+      '';
+
+      dash-exe = "${dash}/bin/dash";
+      psi-plus-exe = "${psi-plus-unwrapped}/bin/psi-plus";
+    in
+    writeTextFile rec {
+      name = "psi-plus";
+      executable = true;
+      destination = "/bin/${name}";
+
+      checkPhase = ''
+        set -eu || exit
+        ${checkExecutableExistence dash-exe}
+        ${checkExecutableExistence psi-plus-exe}
+        ${
+          builtins.concatStringsSep "\n"
+            (map (x: checkDirExistence "${gstLibSubDir x}") gstDependencies)
+        }
+      '';
+
+      text = ''
+        #! ${dash-exe}
+        set -eu || exit
+
+        if [ -n "''${${gstEnvVarName}-}" ]
+        then
+          post=":''$${gstEnvVarName}"
+        else
+          post=
+        fi
+
+        ${gstEnvVarName}=${
+          escapeShellArg
+            ( builtins.concatStringsSep ":"
+                (map (x: "${gstLibSubDir x}") gstDependencies) )
+        }"''$post"
+
+        export -- ${escapeShellArg gstEnvVarName}
+        exec ${escapeShellArg psi-plus-exe} "$@"
```

Comment:
This is a lot of changes. Have you tried an approach used in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/nheko/default.nix and https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/kaidan/default.nix insted?

Reply:
@misuzu thanks, I was looking for something like this. I didn’t know it was possible to patch existing wrapper this way. I force-pushed a change, now it’s only few lines change. Please review.